src/gluonts/nursery/sagemaker_sdk/entry_point_scripts/train_entry_point.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+# Standard library imports
+from pathlib import Path
+import argparse
+import os
+import json
+import logging
+
+# Third-party imports
+
+# First-party imports
+from gluonts.core import serde
+from gluonts.dataset import common
+from gluonts.dataset.repository import datasets
+from gluonts.evaluation import Evaluator, backtest
+
+# Logging: print logs analogously to Sagemaker.
+logger = logging.getLogger(__name__)
+
+
+def train(arguments):
+    """
+    Generic train method that trains a specified estimator on a specified dataset.
+    """
+
+    logger.info("Downloading estimator config.")
+    estimator_config = Path(arguments.estimator) / "estimator.json"
+    with estimator_config.open() as config_file:
+        estimator = serde.load_json(config_file.read())
+
+    logger.info("Downloading dataset.")
+    if arguments.s3_dataset is None:  # == "None":
+        # load built in dataset
+        dataset = datasets.get_dataset(arguments.dataset)
+    else:
+        # load custom dataset
+        s3_dataset_dir = Path(arguments.s3_dataset)
```

Comment:
Not sure I like this distinction between `s3_dataset` and `dataset`.

Reply:
The distinction has to be made, since the built in dataset has to be retrieved first, and the s3_dataset was copied into the container and has to be converted leaded "manually".